app/controllers/subsidy/applications/edit.js

```diff
@@ -2,9 +2,18 @@ import Controller from '@ember/controller';
 import { inject as service } from '@ember/service';
 import { task } from 'ember-concurrency';
 import fetch from 'fetch';
+import { action } from '@ember/object';
+import { saveAs } from 'file-saver';
+import JSZip from 'jszip';
 
 export default class SubsidyApplicationsEditController extends Controller {
   @service router;
+  @service() store;
+
+  constructor() {
+    super(...arguments);
+    window.addEventListener('beforeprint', this.prepareTextareasForPrinting);
```

Comment:
Doesn't this event handler cause issues if they would try to print a different page in the app?

The handler assumes the user is on the edit page, but after visiting the edit page and going to a different route it would still be active AFAIK. I think it might be possible to start / stop the handler from the route's setupController / resetController hooks. Alternatively you could add a new `PrintHandler` component that receives the method as an argument, and do the registration there, and tear it down again in the `willDestroy` method. A custom modifier would also work. 

Multiple options, but I think removing the handler somehow would be the safest?

Reply:
Good catch, the current implementation also kept duplicating when clicking the print multiple times. I added `{ once: true }` and put the addeventlistener of afterprint before calling the window.print(). This should fix it

Reply:
I see! How about we just call the prepare function inside the exportSubsidyAsPDF function?
```

  @action
  async exportSubsidyAsPDF() {
    await this.prepareTextareasForPrinting();
    ...
 ```
 This makes things a lot easier, and we can assume the user will have to click the button anyway to have the right filename.

Reply:
Yea, I think that makes sense!